### PR TITLE
Simplify no case compare

### PR DESCRIPTION
**Pull request by mkiael**
This PR adds the range-v3 library as a dependency. Unfortunately a patch is needed to be able to build the latest release of the library with clang13 and libc++. The patch is a commit from the master branch of the main repo.

Please object if you don't want that we add this dependency or if you have any concern with how the patching is handled.

### CODE REVIEW

File: util/BUILD [hunk 1]
```diff
@@ -7,6 +7,7 @@ cc_library(
         "string.h",
     ],
     visibility = ["//visibility:public"],
+    deps = ["@range-v3//:range-v3"],
```
Comment (robinlinden) [inline util/BUILD:10]: Is it possible to use `@ranges-v3//:std` to ensure that we can switch to the stdlib ranges once they're available in all compilers/stdlib combos we support?

File: WORKSPACE [hunk 2]
```diff
@@ -122,3 +122,12 @@ http_archive(
     strip_prefix = "zlib-1.2.11",
     url = "https://github.com/madler/zlib/archive/v1.2.11.tar.gz",
 )
+
+http_archive(
+    name = "range-v3",
+    patch_args = ["-p1"],
+    patches = ["//third_party/patches:range-v3.patch"],
```
Comment (robinlinden) [inline WORKSPACE:129]: Looking at https://github.com/ericniebler/range-v3/compare/0.11.0...master I think we should just go with the latest commit on master. :P Everything after 0.11.0 appears to be bugfixes.

You can see an example of including an archive of a specifc commit if you look at stb in this file.

Comment (robinlinden) [reply WORKSPACE:129]: I also think I have the libraries in here in alphabetical order (within their group.)

**Review by robinlinden — CHANGES_REQUESTED** [review]

File: util/BUILD (hunk 1 above)
Comment (robinlinden) [inline util/BUILD:10]: This could be simplified to `@ranges-v3`. If the package name matches the dependency's name, it's not needed!

**Review by robinlinden — APPROVED** [review]
Looks great! Looking forward to start using ranges everywhere. :D

**Review by robinlinden — COMMENTED** [review]
Oh, actually the commit message from the deps commit is wrong. It's still talking patching the dependency. :x

**Comment by mkiael** [comment]
Fixed the commit message, good catch!

**Pull request merged**